pumpkin-inventory/src/lib.rs

```diff
@@ -212,26 +223,68 @@ impl<'a> Container for OptionallyCombinedContainer<'a, 'a> {
             .unwrap_or(self.inventory.window_name())
     }
 
-    fn all_slots(&mut self) -> Vec<&mut Option<ItemStack>> {
-        let slots = match &mut self.container {
+    fn iter_slots_mut<'s>(
+        &'s mut self,
+    ) -> Box<(dyn ExactSizeIterator<Item = &mut Option<ItemStack>> + 's)> {
+        match &mut self.container {
             Some(container) => {
-                let mut slots = container.all_slots();
-                slots.extend(self.inventory.all_combinable_slots_mut());
-                slots
+                let container_size = container.size();
+
+                Box::new(ExactSizedChain {
```

Comment:
I think you could extract out this to a function that does the same thing, cause this code appears in like two places rn